### PR TITLE
allow add/remove admins, use emails over subs

### PR DESCRIPTION
**Pull request by susanthomson**
Added server-side functionality to add and remove admins and relevant tests. Rewrote old auth tests to use jasmine over sinon. Admin config file now uses email addresses instead of subs so this will need updated on prod as well as locally.

### CODE REVIEW

File: spec/server/authoriser.spec.js [hunk 1]
```diff
@@ -1,55 +1,69 @@
...
-        };
+        verifier = jasmine.createSpyObj("verifier", [
+            "verify"
+        ]);
 
-        fs = {
-            readFile: function() {}
+        var adminConfig = {
+            emails: ["bob@gmail.com", "alice@gmail.com"]
         };
```
Comment (susanthomson) [inline spec/server/authoriser.spec.js:42]: How the adminConfig should look now ^^

File: spec/server/authoriser.spec.js [hunk 2]
```diff
@@ -1,55 +1,69 @@
...
+        fs = jasmine.createSpyObj("fs", [
+            "readFile",
+            "writeFile",
+            "asdf",
```
Comment (stephen-lt-sl) [inline spec/server/authoriser.spec.js]: :coffin:

File: spec/server/authoriser.spec.js [hunk 3]
```diff
@@ -1,55 +1,69 @@
...
+        ]);
+        fs.readFile.and.callFake(function(file, encoding, callback) {
+            callback(undefined, JSON.stringify(adminConfig));
+        });
+        fs.writeFile.and.callFake(function(file, contents, callback) {
+            console.log("WriteFile called");
```
Comment (stephen-lt-sl) [inline spec/server/authoriser.spec.js]: :zipper_mouth_face:

File: spec/server/authoriser.spec.js [hunk 4]
```diff
@@ -147,4 +162,32 @@ describe("Authoriser", function() {
...
+        var updatedEmails = [];
+        var objToWrite = {};
+
+        it("addAdmin updates admin config if admin is not in list", function() {
+            updatedEmails = ["bob@gmail.com", "alice@gmail.com", "charlie@gmail.com"];
+            objToWrite = {
+                "emails": updatedEmails,
+            };
+            authoriser.addAdmin("charlie@gmail.com").then(function() {
+                console.log("Tests");
```
Comment (stephen-lt-sl) [inline spec/server/authoriser.spec.js]: :scream:

**Review by stephen-lt-sl — APPROVED** [review]
:+1:  pending comments

File: spec/server/authoriser.spec.js (hunk 2 above)
Comment (susanthomson) [reply spec/server/authoriser.spec.js]: looooooool

**Pull request merged**